src/installer/src/main.rs

```diff
@@ -0,0 +1,232 @@
+    }
+}
+
+/// Searches for a file containing "ghidraRun" at provided path recursively.
+fn search_for_ghidrarun(entry_path: &Path) -> Option<PathBuf> {
+    for entry in WalkDir::new(entry_path)
+        .into_iter()
+        .filter_map(|e| e.ok())
+        .filter(|e| e.metadata().unwrap().is_file())
+    {
+        if entry.file_name().to_str().unwrap().contains("ghidraRun") {
+            let mut hit = entry.into_path();
+            hit.pop();
+            return Some(hit);
+        }
+    }
+    None
+}
+
+/// Determines Ghidra versions and provides selection interface for the user.
+fn select_ghidra_version(mut ghidra_locations: Vec<PathBuf>) -> Result<PathBuf> {
+    let mut i = 0;
+    while i < ghidra_locations.len() {
+        ghidra_locations[i].push("Ghidra/application.properties");
+        let app_prop = std::fs::read_to_string(&ghidra_locations[i]);
+
+        match app_prop {
+            Ok(properties) => {
+                let version: &str = properties
+                    .lines()
+                    .filter_map(|x| x.strip_prefix("application.version="))
+                    .next()
+                    .unwrap_or("?");
+                ghidra_locations[i].pop();
+                ghidra_locations[i].pop();
+                println!(
+                    "Use Ghidra at: {} [v{}]? ({})",
+                    ghidra_locations[i].display(),
+                    version,
+                    i
+                );
+                i += 1;
+            }
+
+            Err(_) => {
+                ghidra_locations.remove(i);
+            }
+        }
+    }
+    if ghidra_locations.is_empty() {
+        return Err(Error::new(ErrorKind::NotFound, "Ghidra not found").into());
+    }
+
+    get_user_choice(&ghidra_locations)
+}
+
+/// Determines Ghidra versions and provides selection interface for the user.
+fn get_user_choice(ghidra_locations: &[PathBuf]) -> Result<PathBuf> {
+    println!("Please select (0-{}): ", ghidra_locations.len() - 1);
+
+    let mut choice = String::new();
+
+    std::io::stdin().read_line(&mut choice)?;
+
+    match choice.trim().parse::<usize>() {
+        Ok(i) => {
+            if i < ghidra_locations.len() {
+                Ok(ghidra_locations[i].clone())
+            } else {
+                Err(Error::new(ErrorKind::InvalidInput, "invalid user input").into())
+            }
+        }
+        Err(error) => Err(error.into()),
+    }
+}
+
+/// Creates ghidra.json for a Ghidra location at provided locaton.
+fn create_ghidra_json(location: &Path, ghidra_location: PathBuf) -> Result<()> {
+    let conf = GhidraConfig {
+        ghidra_path: ghidra_location,
+    };
+
+    std::fs::write(location.join("ghidra.json"), serde_json::to_string(&conf)?)?;
+    Ok(())
+}
+
+/// Runs Cargo install to install cwe_checker.
+fn install_cwe_checker() -> Result<()> {
+    std::process::Command::new("cargo")
+        .args(["install", "--path", "src/caller", "--locked"])
+        .status()?;
```

Comment:
When I tried the install script in my Windows VM, this failed (because of something unrelated to your PR), but the installation script did not stop. So apparently the return value of this function was still `Ok(())`. We should investigate why.

Reply:
I identified the problem: `.status()?` checks only that the command finished by unwrapping the exit code of the command. But it does not actually check whether the exit code is zero or not, i.e. we do not check whether it was a success or not. So we also have to check the return value of `.status()?` to make sure that the `cargo install` command was successful.